AndroidAnnotations/androidannotations-rest-spring/rest-spring-test/src/main/java/org/androidannotations/rest/spring/test/MyService.java

```diff
@@ -141,94 +142,92 @@
 	@Post("http://company.com/client/ping")
 	void ping();
 
-	// There should be max 1 parameter that is not mapped to an attribute. This
-	// parameter will be used as the post entity.
 	@Post("/events/")
 	@Accept(MediaType.APPLICATION_JSON)
-	Event addEvent(Event event);
+	Event addEvent(@Body String event);
 
 	@Post("/events/{year}/")
-	Event addEvent(Event event, int year);
+	Event addEvent(@Body Event event, int year);
 
 	@Post("/events/{year}/")
 	Event addEvent(int year);
 
 	@Post("/events/")
-	ResponseEntity<Event> addEvent2(Event event);
+	ResponseEntity<Event> addEvent2(@Body Event event);
 
-	@Post("/events/{date}")
+	@Post("/events/{date}?myCookieInUrl={myCookieInUrl}")
```

Comment:
We only validate the sum, not actual name. You are right. AA has to handle that. I will add the validation in this pull request.